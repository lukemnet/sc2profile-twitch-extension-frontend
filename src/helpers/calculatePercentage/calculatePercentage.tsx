import checkIfZero from 'src/helpers/checkIfZero/checkIfZero';

const calculatePercentage = (number: number, total: number) => {
  const fraction = number / total;
  return checkIfZero(fraction) ? 0 : Math.round(fraction * 100);
};

export default calculatePercentage;
