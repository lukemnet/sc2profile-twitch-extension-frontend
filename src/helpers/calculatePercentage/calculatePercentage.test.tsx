import calculatePercentage from './calculatePercentage';

it('should calculate correct percentages', () => {
  expect(calculatePercentage(0,100)).toMatchSnapshot();
  expect(calculatePercentage(1,150)).toMatchSnapshot();

  expect(calculatePercentage(1354,3434)).toMatchSnapshot();
  expect(calculatePercentage(100,200)).toMatchSnapshot();
  expect(calculatePercentage(100,100)).toMatchSnapshot();
});

it('should return zero for non-positive numbers', () => {
  expect(calculatePercentage(-100,-300)).toBe(0);
  expect(calculatePercentage(-10, 0)).toBe(0);
  expect(calculatePercentage(1,-300)).toBe(0);
});
