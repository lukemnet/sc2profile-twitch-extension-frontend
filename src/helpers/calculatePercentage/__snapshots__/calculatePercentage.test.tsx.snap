// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should calculate correct percentages 1`] = `0`;

exports[`should calculate correct percentages 2`] = `1`;

exports[`should calculate correct percentages 3`] = `39`;

exports[`should calculate correct percentages 4`] = `50`;

exports[`should calculate correct percentages 5`] = `100`;
